Tasks/UseNodeV1/usenode.ts

```diff
@@ -0,0 +1,39 @@
+//
+// UseNode: 
+//     Optionally install version at runtime, setup proxy and setup auth
+//     This allows for natural cmd line steps in yaml after "using" that eco-system
+//     since proxy vars and auth is setup for the rest of the job
+//
+// https://github.com/Microsoft/azure-pipelines-yaml/blob/master/design/use-statement.md
+//
+
+import * as taskLib from 'azure-pipelines-task-lib/task';
+//import * as toolLib from 'vsts-task-tool-lib/tool';
+import * as installer from './installer';
+import * as proxyutil from './proxyutil';
+
+async function run() {
+    try {
+        //
+        // Version is optional.  If supplied, install / use from the tool cache
+        // If not supplied then task is still used to setup proxy, auth, etc...
+        //
+        const version = taskLib.getInput('version', false);
+        if (version) {
+            const checkLatest: boolean = taskLib.getBoolInput('checkLatest', false);
+
+            // TODO: installer doesn't support proxy
```

Comment:
dont we get the proxy settings for free by using the typed rest client?

Reply:
We should. I'll add this, I think we can scope it to a future PR though.